src/stride.rs

```diff
@@ -266,6 +280,21 @@ impl<'a, A> Clone for Stride<'a, A> {
     }
 }
 
+impl<'a, A> StrideMut<'a, A> {
+    /// Return a mutable reference to the element of a stride at the
+    /// given index, or None if the index is out of bounds.
+    pub fn get_mut<'b>(&'b mut self, i: usize) -> Option<&'b mut A> {
+        if i >= self.len() {
+            None
+        } else {
+            unsafe {
+                let ptr = self.begin.offset(self.offset + self.stride * (i as isize));
+                Some(mem::transmute(ptr))
```

Comment:
The macro uses transmute as a lazy way to work with both the mutable and immutable version. This being outside the macro, it can use a regular dereference here instead.

`&mut *ptr` makes a mutable reference from a raw pointer in an `unsafe` block.

I will fix this up when merging.